pkg/issuer/acme/dns/dns.go

```diff
@@ -238,6 +240,29 @@ func (s *Solver) solverForIssuerProvider(providerName string) (solver, error) {
 			providerConfig.AzureDNS.ResourceGroupName,
 			providerConfig.AzureDNS.HostedZoneName,
 		)
+	case providerConfig.RFC2136 != nil:
+		var secret string
+		if len(providerConfig.RFC2136.TSIGSecret.Name) > 0 {
+			tsigSecret, err := s.secretLister.Secrets(s.resourceNamespace).Get(providerConfig.RFC2136.TSIGSecret.Name)
+			if err != nil {
+				return nil, fmt.Errorf("error getting rfc2136 service account: %s", err.Error())
+			}
+			secretBytes, ok := tsigSecret.Data[providerConfig.RFC2136.TSIGSecret.Key]
+			if !ok {
+				return nil, fmt.Errorf("error getting rfc2136 secret key: key '%s' not found in secret", providerConfig.RFC2136.TSIGSecret.Key)
+			}
+			secret = string(secretBytes)
+		}
+
+		impl, err = rfc2136.NewDNSProviderCredentials(
+			providerConfig.RFC2136.Nameserver,
+			string(providerConfig.RFC2136.TSIGAlgorithm),
+			providerConfig.RFC2136.TSIGKey,
+			secret,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error instantiating rfc2136 challenge solver: %s", err.Error())
+		}
```

Comment:
Could you write some test cases for the config parsing code here to the TestSolverFor function.

You can see an example of a test for valid cloudflare configuration here: https://github.com/splashx/cert-manager/blob/425334068025f3d08450ac4952c5263aed90e94b/pkg/issuer/acme/dns/dns_test.go#L158-L188

If you could put a few together, specifically that test different 'secret key ref' keys (i.e. what happens if the user omits the key? do we have a default, or do we return an error that describes the issue?)

Reply:
It's added. Please review the CI infra as the tests are not passing (for some reason I can't control).

Reply:
This has been implemented in `issuer_test.go` and `rfc2136_test.go`